src/docmanager/index.ts

```diff
@@ -219,11 +219,11 @@ interface IModelFactory {
  * The document manager.
  */
 export
-class DocumentManager {
+class DocumentManager<T extends Widget> {
```

Comment:
The `T` represents the types of widgets the widgetFactory will hand back, right?